src/Tester/MembershipTests/SilosStopTests.cs

```diff
@@ -0,0 +1,80 @@
+/*
+Project Orleans Cloud Service SDK ver. 1.0
+ 
+Copyright (c) Microsoft Corporation
+ 
+All rights reserved.
+ 
+MIT License
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and 
+associated documentation files (the ""Software""), to deal in the Software without restriction,
+including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
+subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
+OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Orleans.Runtime;
+using Orleans.TestingHost;
+using UnitTests.GrainInterfaces;
+using UnitTests.Tester;
+
+namespace UnitTests.MembershipTests
+{
+    [DeploymentItem("OrleansConfigurationForSilosStopUnitTests.xml")]
+    [TestClass]
+    public class SilosStopTests : UnitTestSiloHost
+    {
+        public SilosStopTests()
+            : base(new TestingSiloOptions
+            {
+                StartFreshOrleans = true,
+                StartPrimary = true,
+                StartSecondary = true,
+                SiloConfigFile = new FileInfo("OrleansConfigurationForSilosStopUnitTests.xml"),
+
+            })
+        {
+        }
+
+        [ClassCleanup]
+        public static void MyClassCleanup()
+        {
+            StopAllSilos();
+        }
+
+        [TestMethod, TestCategory("Functional"), TestCategory("Liveness")]
+        public async Task SiloUngracefulShutdown_OutstandingRequestsBreak()
+        {
+            var grain = GrainFactory.GetGrain<ILongRunningTaskGrain<bool>>(Guid.NewGuid());
+            var instanceId = await grain.GetRuntimeInstanceId();
+            var target = GrainFactory.GetGrain<ILongRunningTaskGrain<bool>>(Guid.NewGuid());
+            var promise = instanceId.Contains(Primary.Endpoint.ToString()) ?
```

Comment:
Ohh, great!
I did not notice you had ActivationCountBasedPlacement. Awesome , you are testing that featured too!
so then please just add an assert that indeed 2 grains are on different silos. Just assert in the test, to validate this assumption that the rest of the test relies upon.